lib/std/fs.zig

```diff
@@ -2806,8 +2806,9 @@ fn copy_file(fd_in: os.fd_t, fd_out: os.fd_t) CopyFileRawError!void {
             // a 32 bit value so that the syscall won't return EINVAL except for
             // impossibly large files (> 2^64-1 - 2^32-1).
             const amt = try os.copy_file_range(fd_in, offset, fd_out, offset, math.maxInt(u32), 0);
-            // Terminate when no data was copied
-            if (amt == 0) break :cfr_loop;
+            // Terminate as soon as we have copied size bytes or no bytes
+            if (maybe_size != null and maybe_size.? == amt or amt == 0)
```

Comment:
When  #114 is implemented, this will become a compile error due to ambiguous order of operations. Can you use parens here? (I don't even know what zig's order of operations is here without looking it up)

Also the optional part can be improved like this:
```zig
            if (maybe_size) |s| {
                if (s == amt) break :cfr_loop;
            }
```

Reply:
That was my original version, but I wanted to have "one sentence maps to one line" kind of thing. Re-did it with two statements, also getting rid of precedence uncertainty.